homeassistant/components/climate/broadlink.py

```diff
@@ -0,0 +1,414 @@
+import logging
+import json
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from socket import timeout
+from homeassistant.components.climate import ClimateDevice, \
+    SUPPORT_TARGET_TEMPERATURE,  PLATFORM_SCHEMA, \
+    SUPPORT_OPERATION_MODE, STATE_AUTO
+from homeassistant.const import TEMP_CELSIUS, ATTR_TEMPERATURE, CONF_HOST,\
+    CONF_MAC, CONF_NAME, STATE_ON, STATE_OFF
+
+"""
+Support for Chinese wifi thermostats (Floureon, Beok, Beca Energy)
+configuration.yaml
+climate:
+  - platform: broadlink
+    name: xxx
+    mac: xxxx
+    host: xxxx
+"""
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'broadlink'
+REQUIREMENTS = ['broadlink==0.9.0']
+DEPENDENCIES = []
+
+POWER_ON = 1
+POWER_OFF = 0
+AUTO = 1
+MANUAL = 0
+CONF_MODE_LIST = 'modes'
+CONF_MIN_TEMP = 'min_temp'
+CONF_MAX_TEMP = 'max_temp'
+CONF_ADVANCED_CONFIG = 'advanced_config'
+CONF_SCHEDULE_WEEKDAY = 'schedule_week_day'
+CONF_SCHEDULE_WEEKEND = 'schedule_week_end'
+CONF_WEEKDAY = "weekday"
+CONF_WEEKEND = "weekend"
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_MAC): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+    vol.Optional(CONF_MIN_TEMP, default=5): cv.positive_int,
+    vol.Optional(CONF_MAX_TEMP, default=35): cv.positive_int,
+    vol.Optional(CONF_ADVANCED_CONFIG,
+                 default='{"loop_mode": "0", '
+                         '"sen": "0", '
+                         '"osv": "42", '
+                         '"dif": "2", '
+                         '"svh": "35", '
+                         '"svl": "5", '
+                         '"adj": "0", '
+                         '"fre": "01", '
+                         '"pon": "00"}'): cv.string,
+    vol.Optional(CONF_SCHEDULE_WEEKDAY,
```

Comment:
Schedules should not be part of the platform but done using a automation.

Reply:
schedules are parts of the thermostat configuration. This json is send to the thermostat, those use it to switch between night and day temp when on automatic mode.
Automation can be used when the termostat is on manual mode

Reply:
As I said to Danielhiversen :
This thermostat is standalone, and i do not whanted to loose this feature (in case of home assistant, wifi, home router crash, ..., i'd like to keep control over my thermostat 😉

This add more flexibility to the component. If you don't need it, just leave it as-is, and use automation with manual mode on.
If not, this feature allows it.